fendermint/app/src/app.rs

```diff
@@ -490,6 +495,45 @@ where
         Ok(response)
     }
 
+    /// Amend which transactions to put into the next block proposal.
+    async fn prepare_proposal(
+        &self,
+        request: request::PrepareProposal,
+    ) -> AbciResult<response::PrepareProposal> {
+        let txs = request.txs.into_iter().map(|tx| tx.to_vec()).collect();
+
+        let txs = self
+            .interpreter
+            .prepare((), txs)
```

Comment:
Why do we need to pass txs as a parameter to `prepare`? I had the impression that we are creating extra transactions but not modifying?